### ISSUE TITLE
RPi Kernel 6.1.0 uses a different GPIO Layout.

### ISSUE THREAD

**Issue by qnlbnsl** [qnlbnsl/Matrix-IO#4]

**Comment by qnlbnsl** [comment]
Requires changes to 
- matrix-creator-init for scripts that control gpio
- xc3prog... This will be tough. They created a custom loader for the programmer. The programmer needs to be updated.

**Comment by FutureProofHomes** [comment]
I'm running lastest Bookworm OS w/ Linux Kernel 6.1.0-rpi7-rpi-v8 aarch64.  Perhaps the different GPIO layout is the reason why my Matrix Voice hardware can't connect?  https://github.com/gnumpi/esphome_matrixio/issues/14.  Appreciate the possible help.

**Comment by FutureProofHomes** [comment]
> I'm running lastest Bookworm OS w/ Linux Kernel 6.1.0-rpi7-rpi-v8 aarch64. Perhaps the different GPIO layout is the reason why my Matrix Voice hardware can't connect? [gnumpi/esphome_matrixio#14](https://github.com/gnumpi/esphome_matrixio/issues/14). Appreciate the possible help.

Disregard my comment above.  My ESP32 could not connect because I was incorrectly building it's firmware.  This is a separate issue I've resolved.  Thnx!

**Comment by bikeNomad** [comment]
Is there a way I can get the drivers to build and install under any Raspberry Pi OS version?

I have a Raspberry Pi 4 and a Matrix Creator and just need to use it as a microphone array and LED device.

I have tried both Bookworm and Bullseye versions, both armhf and arm64, and have had no success yet.

**Comment by txf-** [comment]
Is there a documented kernel version or raspberry pi os image that can be used, that still supports the the old deprecated GPIO layout?

I can't seem to find an answer if the GPIOs can still be used via sysfs. because if they can maybe it is just a question of modifying these definitions in the constructor?

https://github.com/qnlbnsl/xc3sprog/blob/master/sysfs.cpp

```
IOSysFsGPIO::IOSysFsGPIO(int tms, int tck, int tdi, int tdo)
 : TMSPin(tms), TCKPin(tck), TDIPin(tdi), TDOPin(tdo), tck_fd(-1), tms_fd(-1), tdi_fd(-1), tdo_fd(-1), one("1"), zero("0")
{
  tdi_fd = setup_gpio(TDIPin, 0);
  tms_fd = setup_gpio(TMSPin, 0);
  tck_fd = setup_gpio(TCKPin, 0);
  tdo_fd = setup_gpio(TDOPin, 1);
}

IOSysFsGPIO::~IOSysFsGPIO() {
  unexport_gpio(TDIPin);
  unexport_gpio(TMSPin);
  unexport_gpio(TCKPin);
  unexport_gpio(TDOPin);
}
```

oh, wow. This has been deprecated long before this board was even released...

**Comment by qnlbnsl** [comment]
nope... i'm just maintaining access to the packages. I have not tried to update everything to the new GPIO Layouts.

**Comment by Seeelefant** [comment]
Well, same error here ... you might become famous for fixing it

**Comment by qnlbnsl** [comment]
> Well, same error here ... you might become famous for fixing it

well.... yeah.... it is a bit out of my wheelhouse. the main issue is not the gpio layout or kernel change but the programmer they used to flash the onboard fpga. It requires extensive knowledge on fpga programming to update the library. It is filled with deprecated packages and is just a bit too involved for it to be a non trivial fix.

I would recommend checking out Satellite1 from [FutureProofHomes](https://futureproofhomes.net/). It is basically the matrix creator but with better support and everything is open sourced. with docs and a lot of community support.

**Comment by Seeelefant** [comment]
Well, yes I ´ll do so  - but unfortunately I still have this Matrix Voice. Is there any-maybe old fashioned Raspi-kernel (version ?)-where thi Matrix voice is reliably running?
